test/e2e/operator/deployment_api.go

```diff
@@ -352,7 +486,137 @@ func startOperator(c *deploy.Cluster, d *deploy.Deployment) {
 		return false
 	}, "3m", "1s").Should(BeTrue(), "increase operator deployment replicas to 1")
 
-	framework.Logf("Ensure operator pod is eady.")
+	framework.Logf("Ensure operator pod is ready.")
 	deploy.WaitForOperator(c, d.Namespace)
 	framework.Logf("Operator is restored!")
 }
+
+func switchDeploymentMode(c *deploy.Cluster, f *framework.Framework, depName string, mode api.DeviceMode) api.Deployment {
+	By(fmt.Sprintf("Switching driver mode to '%s'", mode))
+	deployment := deploy.GetDeploymentCR(f, depName)
+	deployment.Spec.DeviceMode = mode
+	deployment = deploy.UpdateDeploymentCR(f, deployment)
+
+	waitForDriverReady(c, f, depName)
+
+	return deployment
+}
+
+// waitForDriverReady checks if all node driver pods are in Ready state
+func waitForDriverReady(c *deploy.Cluster, f *framework.Framework, depName string) {
+	nWorkerNodes := int32(c.NumNodes()) - 1
+
+	// Ensure all the worker driver is ready
+	EventuallyWithOffset(1, func() int32 {
+		ds := c.WaitForDaemonSet(depName + "-node")
```

Comment:
This isn't in `WaitForPMEMDriver`, but I also doubt that it works reliably: if you check immediately after changing the deployment, then it could be that either the PMEM-CSI or the DaemonSet controller have not reacted to the change and thus the DaemonSet is ready because nothing has changed yet.

Reply:
Changed the code to wait till daemonset get updated before calling `WaitForPMEMDriver`.